maven-plugin/src/main/java/net/revelc/code/formatter/FormatterMojo.java

```diff
@@ -319,7 +319,10 @@ public void execute() throws MojoExecutionException, MojoFailureException {
 
         List<File> foundFiles = new ArrayList<>();
         for (String filename : ds.getIncludedFiles()) {
-            foundFiles.add(new File(newBasedir, filename));
+            File foundFile = new File(newBasedir, filename);
+            if(foundFile.canWrite()){
+                foundFiles.add(foundFile);
+            }
```

Comment:
I didn't have a huge amount of time to do the fix.  I had originally added the fix before the call to formatFile(), however, this made the file counts inconsistent.  So I added the check in addCollectionFiles() to prevent readonly files from being added.  For completeness, I left the check around formatFile as if the file list is large there is a chance the status of the file can change from creation of the list to formatting of the file.